packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -997,11 +999,27 @@ class ProjectFileInvalidator {
         printError('Warning: updated dependencies detected. The Flutter application will require a restart to safely use new packages.');
       }
       _packagesUpdateTime = newPackagesUpdateTime;
+      final bool hadBuilders =_hasBuilders;
+      _hasBuilders = _flutterProject?.hasBuilders ?? false;
+      if (hadBuilders != _hasBuilders) {
+        printError('Warning: Added builders detected. The Flutter runner will require a restart to use them.');
```

Comment:
Ah, looks like the case I'm describing above is handled here.

Reply:
I've removed this due to the compiler refactor